gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/usecase/log/SearchConnectionLogUsecase.java

```diff
@@ -0,0 +1,110 @@
+
+    public SearchConnectionLogUsecase(
+        LogStorageService logStorageService,
+        PlanStorageService planStorageService,
+        ApplicationStorageService applicationStorageService
+    ) {
+        this.logStorageService = logStorageService;
+        this.planStorageService = planStorageService;
+        this.applicationStorageService = applicationStorageService;
+    }
+
+    public Response execute(ExecutionContext executionContext, Request request) {
+        var pageable = request.pageable.orElse(new PageableImpl(1, 20));
+
+        var response = logStorageService.searchApiConnectionLog(request.apiId(), pageable);
+        return mapToResponse(executionContext, response);
+    }
+
+    private Response mapToResponse(ExecutionContext executionContext, SearchLogResponse<BaseConnectionLog> logs) {
+        var total = logs.total();
+        var data = logs.logs().stream().map(log -> mapToModel(executionContext, log)).toList();
+
+        return new Response(total, data);
+    }
+
+    private ConnectionLogModel mapToModel(ExecutionContext executionContext, BaseConnectionLog connectionLog) {
+        return ConnectionLogModel
+            .builder()
+            .apiId(connectionLog.getApiId())
+            .requestId(connectionLog.getRequestId())
+            .timestamp(connectionLog.getTimestamp())
+            .application(getApplicationEntity(executionContext, connectionLog.getApplicationId()))
+            .clientIdentifier(connectionLog.getClientIdentifier())
+            .method(connectionLog.getMethod())
+            .plan(getPlanInfo(connectionLog.getPlanId()))
+            .requestEnded(connectionLog.isRequestEnded())
+            .transactionId(connectionLog.getTransactionId())
+            .status(connectionLog.getStatus())
+            .build();
+    }
+
+    private GenericPlanEntity getPlanInfo(String planId) {
+        try {
+            return planStorageService.findById(planId);
+        } catch (PlanNotFoundException | TechnicalManagementException e) {
+            return BasePlanEntity.builder().id(planId).name("Unknown plan").build();
+        }
+    }
+
+    private BaseApplicationEntity getApplicationEntity(ExecutionContext executionContext, String applicationId) {
+        try {
+            return applicationStorageService.findById(executionContext, applicationId);
+        } catch (ApplicationNotFoundException | TechnicalManagementException e) {
+            return BaseApplicationEntity.builder().id(applicationId).name("Unknown application").build();
+        }
+    }
+
+    public record Request(String apiId, String userId, Optional<Pageable> pageable) {
+        public Request(String apiId, String userId) {
+            this(apiId, userId, Optional.empty());
+        }
+        public Request(String apiId, String userId, Pageable pageable) {
+            this(apiId, userId, Optional.of(pageable));
+        }
+    }
+
+    public record Response(long total, List<ConnectionLogModel> data) {}
```

Comment:
Why not, but is it not a little bit over-engineered?

Reply:
:not-sure: about the over-engineered I'm just creating 2 POJO. I'm used to use create a pojo (record in Java or data class in Kotlin) for the input/output usecase. It helps to handle optional parameters without overriding methods.
Also it allows to extract an interface for the usecase but I don't know if we will do that 🙈 

But if it makes everyone mad I can change 🤷